DragaliaAPI.Database/Repositories/FortRepository.cs

```diff
@@ -58,4 +59,64 @@ public async Task<bool> CheckPlantLevel(FortPlants plant, int requiredLevel)
 
         return result;
     }
+
+    public async Task GetFortPlantIdList(IEnumerable<int> fort_plant_id_list)
+    {
+        // What do
+    }
+
+    public async Task<bool> InitFortDetail(string accountId)
+    {
+        await apiContext.PlayerFortDetails.AddAsync(
+            new DbFortDetail()
+            {
+                DeviceAccountId = accountId,
+                CarpenterNum = 2,
+                MaxCarpenterCount = 5,
+                WorkingCarpenterNum = 0
+            }
+        );
+        await apiContext.SaveChangesAsync();
+        return true;
```

Comment:
I think it would be best done that way, yes. As it wasn't done yet, I had to work with what I had for now. It's also the reason why I had to do this:
```
if (!query.Any())
{
    await this.fortRepository.InitFortDetail(this.DeviceAccountId);
    fortDetails = StubData.FortDetail;
}
```
in the Controller, as right now, there's no guarantee that there will be a FortDetail saved for the account on the database. This is merely to ensure that we don't get softlocked when opening the Halidom.

When these changes to the savefile are implemented, this function should be removed entirely.

Reply:
Feel free to change `CreateBase` to do it there, it's not offlimits by any means

Reply:
If a reset or fresh server ever happened, this would no longer be a problem, but I'm unsure if there are plans for that. But to fix the problem for current accounts, the migration would be a good solution.

Maybe this migration and the changes to the SaveFile could be done in a different PR though, as this one is big enough as it is. I would like to keep the band-aid for now, and work on that next, if that's OK with you too.

Reply:
I understand your concerns, but that might make things more complicated. Once this PR is merged in it'll be auto-deployed to the server, so the SQL you write after the fact can't just be a simple INSERT and will have to account for the fact that the data may already exist.

I don't think it is too complicated, something like

```sql
INSERT INTO "PlayerFortDetail" ("DeviceAccountId", "CarpenterNum", "MaxCarpenterCount", "WorkingCarpenterNum")
SELECT "DeviceAccountId", 2, 5, 0 FROM "Players"
```

should work. Additionally, the savefile side would just be moving the code you've already written to a different method.

If you'd like, I am capable of pushing to forks (I think), so once you've finished off your changes I can make this change.

Reply:
In that case, I'll be removing this function from the PR altogether. I also don't like leaving loose ends, so it's probably for the best :)

As for the changes to the savefile, if that works for you, it works for me. I'm still getting to grips with the codebase, so if you already know where the code goes, I'm okay with you taking over that part, no problem.